### ISSUE TITLE
Line # would be nice in these conditions

### ISSUE THREAD

**Issue by ahmadawais** [Otto42/theme-check#153]
Theme Check is an awesome plugin. Thanks for that Otto. But sometimes it can be a pain in ass to find the errors it points out. 

Now E.g. 

`WARNING: Found a translation function that is missing a text-domain. Function _e, with the arguments 'VR'`

I searched everywhere and found no issue. `'VR'` is the text domain, it isn't an argument. If ThemeCheck could point the line where it sees the problem like that, it could be a big help.

**Comment by grappler** [comment]
Do you have the code example so that we can test this false positive?

**Comment by ahmadawais** [comment]
Nops! That is the reason why I submitted this issue. I am unable to find the issue without having a line # attached with the warning. What is your suggestion?

**Comment by grappler** [comment]
Could you share the theme files? You could send them to me if you wanted https://ulrich.pogson.ch/contact-me

**Comment by ahmadawais** [comment]
I am not at liberty to do that. Under a solid NDA contract. But I am going
through all the Text domain instances in hope for finding the on causing
this issue.

**Comment by ahmadawais** [comment]
Isn't there a simple patch to make Theme Check post the line number with
the error?

**Comment by Otto42** [comment]
A minor limitation of the existing i18n checks is that it cannot give you a line number, because it's not reading the code as "lines". Rather, that particular check tokenizes the entire code and reads it in the same way that PHP itself reads it.

"Function _e, with the arguments 'VR'"

This usually means that you made a function call to _e somewhere and gave it a variable as the first parameter. The _e function is not a shortcut for "echo". _e is a translation function, it can only translate *strings*, not *variables*.

Basically, you did this:

_e ( $string, 'VR' );

When you should have done this:

echo $string;

Or something like that.

**Comment by ahmadawais** [comment]
Exactly! I have done this. I never knew that. So, if I plan to share a
variable with text domain I should instead use printf?

**Comment by Otto42** [comment]
Yes, definitely. Please give this post a read. Should help with the fundamentals.

http://ottopress.com/2012/internationalization-youre-probably-doing-it-wrong/

**Comment by ahmadawais** [comment]
@Otto42 and @grappler Thanks a lot for you time. It was all very helpful. Solved the issue with your help. `echo sprintf( __( '%s', 'VR' ), $vr_checkout );`

**Comment by grappler** [comment]
This would still be incorrect as `%s` is the same in all languages. You want to add the translation function further up `$vr_checkout = __( 'Hello', 'VR' )`.

Also you could use `printf()` instead of `echo sprintf()`

**Comment by ahmadawais** [comment]
@grappler Actually that is already taken care of. Here's how the variable def looks like.

```php
// Check out date.
$vr_checkout = isset( $_GET['vr_booking_date_checkout'] ) && ( '' != $_GET['vr_booking_date_checkout'] )
				? $_GET['vr_booking_date_checkout'] : __( 'Check Out', 'VR' );

```

**Comment by Otto42** [comment]
Translating just %s is completely pointless and should not be in any code:

If the $vr_checkout is already being translated elsewhere, then just echo it. Running it through a __ and then sprintf has no value.

**Comment by ahmadawais** [comment]
@Otto42 Sure! Just modified it.
